grakn-engine/src/main/java/ai/grakn/engine/externalcomponents/ExternalComponentException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.externalcomponents;
+
+/**
+ * Should be thrown upon encountering any error when unable to start / stop an external component
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class ExternalComponentException extends RuntimeException {
```

Comment:
This should probably extend `GraknException` or `GraknBackendException` . . . the latter seems like a good one. Check the javadocs of `GraknBackendException` and tell me what you think.